newsrevenuehub/rev-engine
# DEV-2031 | Refactor SocialMeta relationship

Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

This PR is initially into DEV-2030 because it presupposes those changes. Once that PR has been merged, this PR's base should be changed to DEV-2031.

#### What's this PR do?

- Moves the one-to-one between RevenueProgram and SocialMeta to `SocialMeta`, and changes on delete behavior to cascade.
- Migrates existing one-to-ones to the new field on SocialMeta
- Updates admin for RevenueProgram to support changes
- Updates some tests accordingly
- Update some serializers accordingly


#### Why are we doing this? How does it help us?

As part of model relationship cleanup/rationalization described at length in [DEV-1581](https://news-revenue-hub.atlassian.net/jira/software/c/projects/DEV/boards/3?modal=detail&selectedIssue=DEV-1581) and related Google doc.

#### How should this be manually tested?

- Edit the social meta data for a revenue program in the Django admin, and it should work as before
- Open a live page for an RP where socialmeta is setup. Inspect the HTML for the page in dev tools. Inside the head element, you should find the `<meta name="og:<something>" ...>` have values set in admin
  
  <img width="891" alt="Screen Shot 2022-07-08 at 5 14 24 PM" src="https://user-images.githubusercontent.com/960351/178071351-c7c9aa56-c8e1-40a9-a1f7-c6e80ee26516.png">


#### Have automated unit tests been added?

Yes

#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

No

#### Has this been documented? If so, where?

N/A

#### What are the relevant tickets? Add a link to any relevant ones.

[DEV-2031](https://news-revenue-hub.atlassian.net/jira/software/c/projects/DEV/boards/3?modal=detail&selectedIssue=DEV-2031)
[DEV-1581](https://news-revenue-hub.atlassian.net/jira/software/c/projects/DEV/boards/3?modal=detail&selectedIssue=DEV-1581)

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No

#### Are there next steps? If so, what? Have tickets been opened for them?


N/A
